src/ValidatorPluginManager.php

```diff
@@ -589,9 +590,11 @@ public function injectTranslator($first, $second)
             $container = $container->getServiceLocator();
         }
 
-        if ($validator instanceof Translator\TranslatorAwareInterface) {
-            if ($container && $container->has('MvcTranslator')) {
+        if ($validator instanceof Translator\TranslatorAwareInterface && $container instanceof ContainerInterface) {
+            if ($container->has('MvcTranslator')) {
                 $validator->setTranslator($container->get('MvcTranslator'));
+            } elseif ($container->has(TranslatorInterface::class)) {
+                $validator->setTranslator($container->get(Translator\TranslatorInterface::class));
             }
         }
```

Comment:
```suggestion
        if (! $validator instanceof Translator\TranslatorAwareInterface || ! $container instanceof ContainerInterface) {
            return;
        }
        
        if ($container->has('MvcTranslator')) {
            $validator->setTranslator($container->get('MvcTranslator'));
            
            return;
        }
        
        if ($container->has(TranslatorInterface::class)) {
            $validator->setTranslator($container->get(Translator\TranslatorInterface::class));
        }
```

I think that early returns here make the control flow more readable